File: 2025/Day-01/Part01.py
import os

def read_input(filename: str) -> list[str]:
    data: list[str] = []
    with open(filename, 'r') as f:
        data = [line.strip() for line in f.readlines()]
    return data

def dial(position: int, instruction: str, dial_size: int = 100) -> int:
    direction: str = instruction[0]
    distance: int = int(instruction[1:])
    
    if direction == "R":
        position = (position + distance) % dial_size
    elif direction == "L":
        position = (position - distance) % dial_size   
    return position

def main(data: list[str]) -> int:
    position: int = 50
    hits: int = 0
    
    for instruction in data:
        position = dial(position, instruction)
        if position == 0:
            hits += 1
    
    return hits

if __name__ == "__main__":
    input_file: str = os.path.dirname(os.path.abspath(__file__)) + r"\\input.txt"
    data: list[str] = read_input(input_file)
    
    print(f"Part 1: {main(data)}")

File: 2025/Day-01-dart/part01.dart
import 'dart:io';

/// Read and parse input file
List<String> readInput(String filename) {
  return File(filename).readAsLinesSync();
}

int dial(String instruction, int position, {int dialSize = 100}) {
  String direction = instruction[0];
  int distance = int.parse(instruction.substring(1));

  if (direction == "L") {
    position = (position - distance) % dialSize;
  } else if (direction == "R") {
    position = (position + distance) % dialSize;
  }
  return position;
}

int solve(List<String> data) {
  int position = 50;
  int hits = 0;

  for (String instruction in data) {
    position = dial(instruction, position);
    if (position == 0) {
      hits += 1;
    }
  }
  return hits;
}

void main() {
  final scriptDir = File(Platform.script.toFilePath()).parent.path;
  final inputFile = '$scriptDir${Platform.pathSeparator}input.txt';
  final data = readInput(inputFile);

  print("Part 1: ${solve(data)}");
}
